splot/_viz_mpl.py

```diff
@@ -0,0 +1,224 @@
+"""
+Lightweight visualizations for pysal using Matplotlib and Geopandas
+
+TODO
+add Examples
+change function input naming to be in line with bokeh functionality
+"""
+__author__ = ("Stefanie Lumnitz <stefanie.lumitz@gmail.com>")
+
+import matplotlib.pyplot as plt
+import geopandas as gpd
+import pysal as ps
+import splot.plot
+
+from matplotlib import patches, colors
+
+from ._viz_utils import mask_local_auto
+
+
+def lisa_cluster(moran_loc, df, p=0.05, figsize=None, ax=None,
+                 legend=True, legend_kwds=None):
+    """
+    Create a LISA Cluster map
+    
+    Parameters
+    ----------
+    moran_loc : esda.moran.Moran_Local instance
+        Values of Moran's Local Autocorrelation Statistic
+    df : geopandas dataframe instance, optional
+        If given, assign df['labels'] per row.  Note that `df` will be
+        modified, so calling functions should use a copy of the user
+        provided `df`.
+    p : float, optional
+        The p-value threshold for significance. Points will
+        be colored by significance.
+    figsize: tuple, optional
+        W, h of figure. Default = None
+    ax : matplotlib Axes instance, optional
+        Axes in which to plot the figure in multiple Axes layout. Default = None
+    legend : boolean, optional
+        If True, legend for maps will be depicted. Default = True
+    legend_kwds : dict, optional
+        Dictionary to control legend formatting options.
+        Example ``legend_kwds={'loc': 'upper left', 'bbox_to_anchor': (0.92, 1.05)}``
+        Default = None
+        
+    Returns
+    -------
+    fig : matplotlip Figure instance
+        Figure of LISA cluster map
+    
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt
+    >>> import libpysal.api as lp
+    >>> from libpysal import examples
+    >>> import geopandas as gpd
+    >>> import esda
+    >>> from splot.mpl import lisa_cluster
+
+    >>> link = examples.get_path('columbus.shp')
+    >>> df = gpd.read_file(link)
+    >>> y = df['HOVAL'].values
+    >>> w = lp.Queen.from_dataframe(df)
+    >>> w.transform = 'r'
+    >>> moran_loc = esda.moran.Moran_Local(y, w)
+
+    >>> fig = lisa_cluster(moran_loc, df)
+    >>> plt.show()
+    """
+    # retrieve colors5 and labels from mask_local_auto
+    _, colors5, _, labels = mask_local_auto(moran_loc, p=p, df=df)
+    
+    # define ListedColormap
+    hmap = colors.ListedColormap(colors5)
+    
+    if ax is None:
+        fig, ax = plt.subplots(1, figsize=figsize)
+    else:
+        fig = ax.get_figure()
+    
+    df.assign(cl=labels).plot(column='cl', categorical=True,
+        k=2, cmap=hmap, linewidth=0.1, ax=ax,
+        edgecolor='white', legend=legend, legend_kwds=legend_kwds)
+    ax.set_axis_off()
+    return fig
```

Comment:
I personally think it can be more useful to return the `ax`, because often you need the ax to further modify the figure (although of course some things like saving the figure rather needs the `fig` object ..) 
This is also what geopandas does (and I think eg also seaborn)

Reply:
I changed the returns for `lisa_cluster()` to `fig, ax`. I think this should be part of the discussion about the splot API, so I added this here: https://github.com/pysal/splot/issues/9